packages/wrapper/src/wrapper.ts

```diff
@@ -0,0 +1,184 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) 2024 TypeFox GmbH (http://www.typefox.io). All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import { editor } from 'monaco-editor';
+import { MonacoLanguageClient } from 'monaco-languageclient';
+import { checkServiceConsistency, configureServices } from './vscode/services.js';
+import { EditorAppExtended, EditorAppConfigExtended } from './editorAppExtended.js';
+import { EditorAppClassic, EditorAppConfigClassic } from './editorAppClassic.js';
+import { ModelUpdate } from './editorAppBase.js';
+import { LanguageClientConfig, LanguageClientWrapper } from './languageClientWrapper.js';
+import { Logger, LoggerConfig } from './logger.js';
+import { InitializeServiceConfig, initServices } from 'monaco-languageclient/vscode/services';
+
+export type WrapperConfig = {
+    serviceConfig?: InitializeServiceConfig;
+    editorAppConfig: EditorAppConfigExtended | EditorAppConfigClassic;
+};
+
+export type UserConfig = {
+    id?: string;
+    loggerConfig?: LoggerConfig;
+    wrapperConfig: WrapperConfig;
+    languageClientConfig?: LanguageClientConfig;
+}
+
+/**
+ * This class is responsible for the overall ochestration.
+ * It inits, start and disposes the editor apps and the language client (if configured) and provides
+ * access to all required components.
+ */
+export class MonacoEditorLanguageClientWrapper {
+
+    private id: string;
+
+    private editorApp: EditorAppClassic | EditorAppExtended | undefined;
+    private languageClientWrapper?: LanguageClientWrapper;
+    private logger: Logger;
+    private initDone = false;
+
+    /**
+     * Perform an isolated initialization of the user services and the languageclient wrapper (if used).
+     */
+    async init(userConfig: UserConfig) {
+        if (this.initDone) {
+            throw new Error('init was already performed. Please call dispose first if you want to re-start.');
+        }
+        if (userConfig.wrapperConfig.editorAppConfig.useDiffEditor && !userConfig.wrapperConfig.editorAppConfig.codeOriginal) {
+            throw new Error('Use diff editor was used without a valid config.');
```

Comment:
Can this error message be more elaborate that the `codeOriginal` attribute in the config?

Reply:
I will extend the message. Will come with next commit.